### DISCUSSION TITLE
`EnableReadOperation` to `rueidisprob.BloomFilter`

### DISCUSSION THREAD

**Discussion by proost** [redis/rueidis#597] [Ideas]
# Problems:
I found that when using `BloomFilter` we can't use replica. 

Because `BITFIELD` is `write` category command. 
we can't simply change to `BITFIELD` to `BITFIELD_RO`. because `BITFIELD_RO` supports from 6.0.0 version. change narrow down supported redis version to use `rueidisprob`.

Event if we change `BITFIELD` command to `BITFIELD_RO`, it is difficult to use replica. Because we always use `EVAL` and `EVALSHA` command to read & write. For example if we use `SendToReplicas` option for read-write splitting, we can't easily configure `SendToReplicas` due to hard to distinguish read & write commands.

# Solution

Adding 

```
// BloomFilterOptions is used to configure BloomFilter.
type BloomFilterOptions struct {
	enableReadOperation bool
}

// BloomFilterOptionFunc is used to configure BloomFilter.
type BloomFilterOptionFunc func(*BloomFilterOptions)

// WithEnableReadOperation enables read operation.
// If enabled, Exists and ExistsMulti methods will be available as read-only operations.
// NOTE: If enabled, minimum redis version should be 7.0.0.
func WithEnableReadOperation(enableReadOperations bool) BloomFilterOptionFunc {
	return func(o *BloomFilterOptions) {
		o.enableReadOperation = enableReadOperations
	}
}
``` 

we add options to constructor like this

```
func NewBloomFilter(
	client rueidis.Client,
	name string,
	expectedNumberOfItems uint,
	falsePositiveRate float64,
	opts ...BloomFilterOptionFunc,
) (BloomFilter, error) {
```

So if `enableReadOperation` is true, we use `BITFIELD_RO`, `EVAL_RO`, `EVALSHA_RO`.

## Considering
I don't want to change minimum version of `rueidisprob.BloomFilter` to 6.0.0 or 7.0.0

**Comment by rueian** [comment]
Great!

**Comment by proost** [comment]
https://github.com/redis/rueidis/pull/598

Closed due to above PR
